app/controllers/application_controller.rb

```diff
@@ -4,7 +4,147 @@ class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
-    set :views, 'app/views'
+    # set :views, 'app/views'
+    set :views, Proc.new { File.join(root, "../views/") }
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    end 
+    erb :'users/create_user'
+  end
+
+  post '/signup' do
+    unless params[:username].empty? || params[:email].empty? || params[:password].empty?
+      user = User.new(username: params[:username], email: params[:email], password: params[:password] )
+      user.save
+      session[:user_id] = user.id
+      redirect '/tweets'
+    else
+      redirect '/signup'
+    end
+  end
+
+  get '/tweets' do
```

Comment:
This controller is overloaded. Consider separating your concerns by splitting some of these actions into separate controllers (i.e. a users controller and a tweets controller).